test/test_container_application.py

```diff
@@ -0,0 +1,194 @@
+        s2i_args=f"--pull-policy=never {container_args} {container_lib.build_s2i_npm_variables()}",
+        src_image=VARS.IMAGE_NAME,
+        dst_image=f"{VARS.IMAGE_NAME}-{app_name}"
+    )
+    return s2i_app
+
+
+class TestPHPTestApplicationContainer:
+    def setup_method(self):
+        self.s2i_app = build_s2i_app(test_app)
+
+    def teardown_method(self):
+        self.s2i_app.cleanup()
+
+    def test_run_s2i_usage(self):
+        output = self.s2i_app.s2i_usage()
+        assert output
+
+    def test_docker_run_usage(self):
+        assert PodmanCLIWrapper.call_podman_command(
+            cmd=f"run --rm {VARS.IMAGE_NAME} &>/dev/null",
+            return_output=False
+        ) == 0
+
+    @pytest.mark.parametrize(
+        "container_arg",
+        [
+            "",
+            "-u 12345",
+        ]
+    )
+    def test_run_app_test(self, container_arg):
+        """
+        Test checks if php-version is proper one
+        and specific pages returns expected message or return code.
+        """
+        cid_file_name = self.s2i_app.app_name
+        assert self.s2i_app.create_container(
+            cid_file_name=cid_file_name, container_args=f"--user=100001 {container_arg}"
+        )
+        assert ContainerImage.wait_for_cid(cid_file_name=cid_file_name)
+        cid = self.s2i_app.get_cid(cid_file_name=cid_file_name)
+        assert cid
+        assert VARS.VERSION in PodmanCLIWrapper.podman_exec_shell_command(
+            cid_file_name=cid,
+            cmd="php --version"
+        )
+        assert VARS.VERSION in PodmanCLIWrapper.podman_exec_shell_command(
+            cid_file_name=cid,
+            cmd="php --version",
+            used_shell="/bin/sh"
+        )
+
+        cip = self.s2i_app.get_cip(cid_file_name=cid_file_name)
+        assert cip
+        # This function tests PHP session
+        assert self.s2i_app.test_response(
+            url=cip, page="/session_test.php",
+            expected_output="Passed"
+        )
+        # Assert tests if connect to http://<ip>:8080 works
+        assert self.s2i_app.test_response(
+            url=cip
+        )
+        # Assert tests if connect to https://<ip>:8443 works
+        assert self.s2i_app.test_response(
+            url=f"https://{cip}", port=8443
+        )
+        # test if configuration is writeable for php.ini
+        assert PodmanCLIWrapper.podman_run_command_and_remove(
+            cid_file_name=VARS.IMAGE_NAME,
+            cmd="[ -w \\${PHP_SYSCONF_PATH}/php.ini ]",
+            return_output=False
+        ) == 0
+        # test if php.d directory is writeable
+        assert PodmanCLIWrapper.podman_run_command_and_remove(
+            cid_file_name=VARS.IMAGE_NAME,
+            cmd="[ -w \\${PHP_SYSCONF_PATH}/php.d ]",
+            return_output=False
+        ) == 0
+
+
+class TestPHPClearEnvContainer:
+    def setup_method(self):
+        container_args = "-e PHP_CLEAR_ENV=OFF" if VARS.OS in ("rhel9", "rhel10") else ""
```

Comment:
If you reverse this condition, you won't have to add new RHEL version every time. This way, if nothing changes, something will probably break with RHEL 11 but if you write it like `"" if VARS.OS == "rhel8" else "-e PHP_CLEAR_ENV=OFF"` you will not need to change it in the future, and when we drop support for RHEL 8, it's gonna be easier to find.